pkg/kubelet/kubelet_pods.go

```diff
@@ -886,12 +887,17 @@ func (kl *Kubelet) getPullSecretsForPod(pod *v1.Pod) []v1.Secret {
 		secret, err := kl.secretManager.GetSecret(pod.Namespace, secretRef.Name)
 		if err != nil {
 			klog.InfoS("Unable to retrieve pull secret, the image pull may not succeed.", "pod", klog.KObj(pod), "secret", klog.KObj(secret), "err", err)
+			failedPullSecrets = append(failedPullSecrets, secretRef.Name)
 			continue
 		}
 
 		pullSecrets = append(pullSecrets, *secret)
 	}
 
+	if len(failedPullSecrets) > 0 {
+		kl.recorder.Eventf(pod, v1.EventTypeWarning, "FailedToRetrieveImagePullSecret", "Unable to retrieve image pull secrets %s; attempting to pull the image may not succeed.", strings.Join(failedPullSecrets, ", "))
```

Comment:
I recognize that this message started out as a copy of the one above at line 889. It doesn't read well for the case when there's only one _Secret_ that we couldn't fetch, as the word "secrets" is still pluralized when it should not be. Rather than trying to correct that word conditionally based on the _Secret_ count, consider the following rewording:
```suggestion
		kl.recorder.Eventf(pod, v1.EventTypeWarning, "FailedToRetrieveImagePullSecret", "Unable to retrieve some image pull secrets (%s); attempting to pull the image may not succeed.", strings.Join(failedPullSecrets, ", "))
```

Reply:
Yeah, the message was indeed taken from the info log but I agree that this case is different since there might be more than one. Thanks @seh I'll fix it right away 😊